File: app/src/main/java/com/casasw/bankapp/CurrencyConfigurator.java
package com.casasw.bankapp;

import java.lang.ref.WeakReference;


public enum CurrencyConfigurator {
    INSTANCE;

    public void configure(CurrencyActivity activity) {

        CurrencyRouter router = new CurrencyRouter();
        router.activity = new WeakReference<>(activity);

        CurrencyPresenter presenter = new CurrencyPresenter();
        presenter.output = new WeakReference<CurrencyActivityInput>(activity);

        CurrencyInteractor interactor = new CurrencyInteractor();
        interactor.output = presenter;

        if (activity.output == null) {
            activity.output = interactor;
        }
        if (activity.router == null) {
            activity.router = router;
        }
    }
}

File: BankApp/app/src/main/java/com/schaefer/bankapp/currency_screen/CurrencyConfigurator.kt
package com.schaefer.bankapp.currency_screen

import com.schaefer.bankapp.currency_screen.activity.CurrencyActivity
import com.schaefer.bankapp.login_screen.presenter.CurrencyPresenterImp
import com.schaefer.bankapp.login_screen.router.CurrencyRouter
import java.lang.ref.WeakReference

enum class CurrencyConfigurator {
    INSTANCE;

    fun configure(activity: CurrencyActivity) {
        val router = CurrencyRouter()
        router.activity = WeakReference<CurrencyActivity>(activity)

        val presenter = CurrencyPresenterImp(activity, activity.applicationContext)
        presenter.currencyActivityInput = activity

        if (activity.router == null) {
            activity.router = router
        }

        if (activity.presenter == null) {
            activity.presenter = presenter
        }
    }
}